src/iterators.jl

```diff
@@ -0,0 +1,199 @@
+abstract type ExpressionIterator end
+
+
+"""
+    ExpressionIterator(grammar::Grammar, max_depth::Int, sym::Symbol)
+An iterator over all possible expressions of a grammar up to max_depth with start symbol sym.
+Types of search depends on the order of production rules in the given grammar: BFS - terminals come first; DFS: nonterminals come first
+"""
+mutable struct ContextFreeIterator <: ExpressionIterator
+    grammar::Grammar
```

Comment:
This should probably be a `ContextFreeGrammar` instead of a `Grammar` for the `ContextFreeIterator`